### ISSUE TITLE
AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'assign' when using model as layer

### ISSUE THREAD

**Issue by markusforster** [keras-team/keras#13607]
**System information**  
- Have I written custom code (as opposed to using example directory):  no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Ubuntu 18.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  tensorflow-gpu 2.0.0
- Keras version:  2.3.0
- Python version:  3.6.8
- CUDA/cuDNN version:  10.2
- GPU model and memory:   GeForce RTX 2080 8GB

I created the following model for testing:

```python
inputLayer1 = Input(shape=(maxWordLen,),name='wordInput')
embedding1 = Embedding(self.numOfChar, 16, input_length=maxWordLen,embeddings_initializer='random_uniform')(inputLayer1)
output1 = Conv1D(4,3,activation='relu',kernel_initializer='random_uniform')(embedding1)
output1 = MaxPooling1D(2)(output1)
output1 = GlobalMaxPooling1D()(output1)
            
wordEncoder = Model(inputLayer1,output1,name='wordEncoder')
            
inputLayer2 = Input(shape=(maxWordCount,maxWordLen),name='mainInput')
lstmIn = TimeDistributed(wordEncoder)(inputLayer2)
            
lstm1 = Bidirectional(LSTM(32, return_sequences=True,  dropout=dropout,recurrent_dropout=dropout))(lstmIn)
lstm2 = Bidirectional(LSTM(32, return_sequences=True,  dropout=dropout,recurrent_dropout=dropout))(lstmIn)

outputLayer2 = add([lstm1,lstm2])
outputLayer2 = TimeDistributed(Dense(numOfLabel, activation='softmax'))(outputLayer2)
sentEncoder = Model(inputLayer2,outputLayer2)

sentEncoder.compile(loss=loss,optimizer=optimizer,metrics=metrics)
```

When running `.fit` on this model I receive the following error:
```python
AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'assign'
```

Not sure if I'm doing it wrong or if it's a problem with tensorflow 2.

**Comment by markusforster** [comment]
Figured out that my problem is in keras/backend/tensorflow_backend.py in function
```python
def zeros(shape, dtype=None, name=None)
```
changed line 963 to:
```python
v = tf.Variable(np.zeros(shape=shape, dtype=dtype), name=name)
```
in order to make it a variable instead of a Tensor Object as is returned with tf.zeros.

**Comment by anurag0406** [comment]
Markus, i got the same error as but i cannot understand how to resolve it, I am new to ML, can you help me out with step by step procedure in layman language

**Comment by anurag0406** [comment]
> Figured out that my problem is in keras/backend/tensorflow_backend.py in function
> 
> ```python
> def zeros(shape, dtype=None, name=None)
> ```
> 
> changed line 963 to:
> 
> ```python
> v = tf.Variable(np.zeros(shape=shape, dtype=dtype), name=name)
> ```
> 
> in order to make it a variable instead of a Tensor Object as is returned with tf.zeros.

Markus, i got the same error as but i cannot understand how to resolve it, I am new to ML, can you help me out with step by step procedure in layman languag
